tools/tasks.py

```diff
@@ -149,6 +149,7 @@ def __init__(self, description, callable, timeout=None, **kwargs):
         self.callable = callable
         self.command = "pythontask"
         self.timeout = timeout
+        self.log_exception = False  # default to false, may be overridden by val in **kwargs
```

Comment:
Would it not make more sense to move this to class level, like the others that are overridable with kwargs?

Reply:
you mean a "static class variable" associated w/ the class rather than the instance?  

Right now the current callers of PythonTask and `make_curl_task` allows some code to enable exception logging and others to keep it disabled.  Whether or not we actually _need_ that level of flexibility could be debated, and maybe just a single module level value that was controlled by a CLI arg would be sufficient.

But that change would touch a lot of code, and potentially cause more bugs, and this repo has 0 test coverage .. so I think a change like that would be better done as part of a larger change (aka "rewrite") that added test coverage or was at least going to go through a solid round of testing.

Reply:
Nah, not a static class variable, I just mean it feels more natural to instantiate `log_exception` alongside `num_samples` and `interval`. Something like `foo` here:

```
>>> class MyClass():
...     foo = 1
...     def __init__(self, bar, **kwargs):
...             self.bar = bar
...             self.__dict__.update(kwargs)
...
>>> cl1 = MyClass('one')
>>> cl1.foo
1
>>> cl1.bar
'one'
>>> cl2 = MyClass('two', foo=2)
>>> cl2.foo
2
>>> cl2.bar
'two'
>>> cl1.foo
1
>>> cl1.bar
'one'
```

Arguably, `self.command = "pythontask"` could be moved up too - everything else we set in `__init__` is an explicit, named variable. Tbh, it's a bit of a moot point, but it just feels neater to me.

Reply:
In that case, `foo` is a class variable rather than an instance variable:

```
$ python
>>> class MyClass():
...     foo = 1
...
>>> MyClass.foo
1
```

Reply:
In your example you're layering instance variables on top of class variables.

I looked it up in the Learning Python O'Reilly book, chapter 29 "Class Coding Details"

Reply:
Yeah, good point, not static though so can be changed by each instantiation, right? Truth be told, I'm probably just nitpicking ;), can't see any major issue with how it currently is.

Reply:
Yeah, an instance variable will "overlay" on top of a class variable.  In your example:

```
>>> cl2.foo
2
```

yet the underlying class variable remains unchanged:

```
>>> MyClass.foo
1
```